packages/block-library/src/list/v2/migrate.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createBlock } from '@wordpress/blocks';
+
+function createListBlockFromDOMElement( listElement ) {
+	const listAttributes = {
+		ordered: 'OL' === listElement.tagName,
+		start: listElement.getAttribute( 'start' )
+			? parseInt( listElement.getAttribute( 'start' ), 10 )
+			: undefined,
+		reversed:
+			listElement.getAttribute( 'reversed' ) === true ? true : undefined,
+		type: listElement.getAttribute( 'type' ) ?? undefined,
+	};
+
+	const innerBlocks = Array.from( listElement.children ).map(
+		( listItem ) => {
+			const children = Array.from( listItem.childNodes );
+			children.reverse();
+			const [ nestedList, ...nodes ] = children;
```

Comment:
Ideally, the function createListBlockFromDOMElement would also be used for raw handling of HTML we are in the presence of the same situation passing from dom nodes to blocks. In that case we can not assume nestedList is the last element in practice in many cases we have this string "<ul>\n<li>sdfdsf</li>\n</ul>\n" the last element is a textNode with "\n". A simple solution would be to ignore nodes after the UL/OL.

Reply:
I think raw handling is a bit different in the sense that it's supposed to deal with any kind of random markup, while this function is especially tailored towards the "saved" content of list block v1 which is less likely to have a list in a middle or something. That said, protecting against empty nodes seems worth it.